### PR TITLE
Job exit when all nodecheck failed

### PR DESCRIPTION
**Pull request by majieyue**
### What changes were proposed in this pull request?

exit job if all workers can't pass node check

### Why are the changes needed?

no need to re-check or try to fault tolerant this error

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

File: dlrover/python/master/node/dist_job_manager.py [hunk 1]
```diff
@@ -243,7 +243,42 @@ def is_all_reduce_type_job(self):
...
+                "Stop the training early because all worker nodes has "
+                "failed the node check in rendezvous."
+            )
+
+            self._process_error(
+                None,
+                0,
+                msg,
+                level=TrainingExceptionLevel.RDZV_ERROR,
+            )
```
Comment (samplise) [inline dlrover/python/master/node/dist_job_manager.py:270]: process_error(message=msg, level=...)

File: dlrover/python/master/node/dist_job_manager.py [hunk 2]
```diff
@@ -243,7 +243,42 @@ def is_all_reduce_type_job(self):
             == DistributionStrategy.ALLREDUCE
         )
 
+    def is_all_workers_node_check_failed(self):
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/dist_job_manager.py:246]: Better move this function into ```WorkerManager```. And work for 'AllReduceType' training only.

**Review by BalaBalaYi — CHANGES_REQUESTED** [review]

File: dlrover/python/master/node/dist_job_manager.py [hunk 3]
```diff
@@ -243,7 +243,42 @@ def is_all_reduce_type_job(self):
...
+        return all(
+            [
+                node.is_node_check_failed()
+                for _, node in self._job_nodes[NodeType.WORKER].items()
```
Comment (workingloong) [inline dlrover/python/master/node/dist_job_manager.py:250]: `for node in self._job_nodes[NodeType.WORKER].values()`

File: dlrover/python/master/node/worker.py [hunk 4]
```diff
@@ -556,3 +556,7 @@ def get_nodes_timeout(self) -> int:
 
     def update_node_required_info(self, nodes_required: Tuple[int, int, int]):
         self._nodes_required = nodes_required
+
+    def is_all_workers_node_check_failed(self):
+        nodes = self._job_context.job_nodes_by_type(NodeType.WORKER)
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/worker.py:561]: ```nodes = self._get_nodes()```

Comment (majieyue) [reply dlrover/python/master/node/worker.py:561]: done

**Review by BalaBalaYi — APPROVED** [review]
lgtm

**Pull request merged**